### PR TITLE
Validations

### PR DESCRIPTION
**Pull request by avneetmalhotra**

### CODE REVIEW

File: app/views/products/_form.html.erb [hunk 1]
```diff
@@ -31,6 +31,21 @@
...
   </div>
 
+  <div class="field">
+    <%= form.check_box :enabled %>
+    <%= form.label :enabled %>
+  </div>
+
+  <div>
+    <%= form.label :discount_price %>
+    <%= form.text_field :discount_price, :placeholder => 'Discount Price' %>
```
Comment (nimish13) [inline app/views/products/_form.html.erb]: please use new hash syntax

File: app/models/product.rb [hunk 2]
```diff
@@ -1,18 +1,44 @@
...
   before_destroy :ensure_not_referenced_by_any_line_item
 
   #...
-  validates :title, :description, :image_url, presence: true
-  validates :price, numericality: {greater_than_or_equal_to: 0.01}
-  validates :title, uniqueness: true
-  validates :image_url, allow_blank: true, format: {
-    with: %r{\.(gif|jpg|png)\Z}i,
-    message: 'must be a URL for GIF, JPG or PNG image.'
+  validates :title, :description, :image_url, :price, :permalink, :discount_price, presence: true
```
Comment (nimish13) [inline app/models/product.rb]: don't force user to enter discount price

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -73,6 +73,6 @@ def set_user
 
     # Never trust parameters from the scary internet, only allow the white list through.
     def user_params
-      params.require(:user).permit(:name, :password, :password_confirmation)
+      params.require(:user).permit(:name,:email, :password, :password_confirmation)
```
Comment (nimish13) [inline app/controllers/users_controller.rb]: spacing issue

**Comment by avneetmalhotra** [comment]
closing in favour of #3

**Pull request closed without merging**